src/modules/pushPair/PushPairModule.ts

```diff
@@ -70,6 +77,14 @@ export class PushPairModule extends Module {
                     return null;
                 }
 
+                // When the prices don't deviate too much we don't need to update the price pair
+                if(!shouldPricePairUpdate(unresolvedRequest, lastUpdate, new Big(outcome.answer), this.prices.get(unresolvedRequest.internalId))) {
+                    logger.debug(`[${this.id}] ${unresolvedRequest.internalId} Price ${outcome.answer} doesn't deviate ${unresolvedRequest.extraInfo.deviationPercentage}% from ${this.prices.get(unresolvedRequest.internalId)}`);
+                    remainingInterval = this.internalConfig.interval;
+                    return null;
+                }
+
+                this.prices.set(unresolvedRequest.internalId, new Big(outcome.answer));
```

Comment:
Maybe a comment on the limitation on setting the price before pushing the price to the contract (which may fail).
We discussed, this would required another approach such as using callbacks. :smile: